### PR TITLE
MiLoG day time exceedance check

### PR DESCRIPTION
**Pull request by gstuer**
As stated by #59, we received the information that the maximum working time per day is legally limited to 10 hours. As a consequence, the following major changes are made by this PR:

- Added a checker method for daily time exceedance
- Added a specific error message for this new method
- Added a JUnit test class to validate the new method

Resolves: #59

### CODE REVIEW

**Comment by larsk21** [comment]
You said you added a test class for the new day time exceedence test, but it doesn't seem to show up in the PR. Or am I missing something?

File: src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+import org.junit.Test;
+
+import data.Employee;
+import data.Entry;
+import data.Profession;
+import data.TimeSheet;
+import data.TimeSpan;
+import data.WorkingArea;
+
+public class MiLoGCheckerDayTimeExceedanceTest {
```
Comment (larsk21) [inline src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java:19]: Is the check successful when there are zero entries on the timesheet?

File: src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        Entry[] entries = {entry};
+        TimeSheet timeSheet = new TimeSheet(EMPLOYEE, PROFESSION, YEAR_MONTH, entries, ZERO_TS, ZERO_TS, ZERO_TS);
+        MiLoGChecker checker = new MiLoGChecker(timeSheet);
+
+        ////Execution
+        checker.checkDayTimeExceedance();
+
+        ////Expectation
+        String error = String.format(MiLoGChecker.CheckerErrorMessage.DAY_TIME_EXCEEDANCE.getErrorMessage(), 
+                MiLoGChecker.getWorkdayMaxWorkingTime(), WORKINGDAY_VALID);
```
Comment (larsk21) [inline src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java:51]: This not only tests the check method, but also the construction of the error message. I don't know if there is a better solution for now, but I'm unhappy with the error construction in general, so maybe we should address that in a separate PR.

File: src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    @Test
+    public void testMultipleExceedances() throws CheckerException {
+        ////Test values
+        TimeSpan start1 = new TimeSpan(9, 59);
+        TimeSpan end1 = new TimeSpan(20, 0);
+        TimeSpan start2 = new TimeSpan(15, 0);
+        TimeSpan end2 = new TimeSpan(20, 0);
+        TimeSpan start3 = new TimeSpan(11, 0);
+        TimeSpan end3 = new TimeSpan(21, 1);
+        LocalDate[] validWorkingDays = {LocalDate.of(2019, 11, 21), WORKINGDAY_VALID,LocalDate.of(2019, 11, 21)};
```
[the remark this answers is not in the document]
Comment (gstuer) [reply src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java:130]: No, this is a mistake! Will be fixed :)

File: src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    
+    @Test
+    public void testRandomSingleEntry() throws CheckerException {
+        ////Random
+        Random rand = new Random();
+
+        ////Test values
+        TimeSpan end = new TimeSpan(rand.nextInt(RANDOM_HOUR_BOUND - 1) + 1, rand.nextInt(RANDOM_MINUTES_BOUND));
+        TimeSpan start = new TimeSpan(rand.nextInt(end.getHour()), rand.nextInt(RANDOM_MINUTES_BOUND));
+
```
[the remark this answers is not in the document]
Comment (gstuer) [reply src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java:238]: Yeah, you're right. This trade-off was made to simplify the logic of the test. Let me see if I can find a more precise solution without making the test much more complicated to understand and validate.

File: src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java (hunk 1 above)
Comment (gstuer) [reply src/test/java/checker/MiLoGCheckerDayTimeExceedanceTest.java:19]: Of course it should be, but I will add this test in order to prove this behaviour.

File: src/main/java/checker/MiLoGChecker.java [hunk 5]
```diff
@@ -96,15 +98,42 @@ protected void checkTotalTimeExceedance() {
...
+                // If so, the working times are summed up and written back to the map.
+                TimeSpan summedTime = workingTimeMap.get(entry.getDate()).add(entry.getWorkingTime());
+                workingTimeMap.put(entry.getDate(), summedTime);
+            } else {
+                // If not, a new entry will be created in the map.
+                workingTimeMap.put(entry.getDate(), entry.getWorkingTime());
+            }
+        }
+        
+        for (LocalDate workingDay : workingTimeMap.keySet()) {
```
[the remark this answers is not in the document]
Comment (gstuer) [reply src/main/java/checker/MiLoGChecker.java:124]: Good Point! I will try to shorten this code segment a little bit with this solution.

**Pull request merged**